portality/api/v1/common.py

```diff
@@ -9,4 +10,30 @@ def default(self, o):
         return o.data
 
 def jsonify_models(models):
-    return app.response_class(json.dumps(models, cls=ModelJsonEncoder), 200, {'Access-Control-Allow-Origin': '*'}, mimetype='application/json')
\ No newline at end of file
+    data = json.dumps(models, cls=ModelJsonEncoder)
+    return respond(data, 200)
+
+def respond(data, status):
+    callback = request.args.get('callback', False)
+    if callback:
+        content = str(callback) + '(' + str(data) + ')'
+        return app.response_class(content, status, {'Access-Control-Allow-Origin': '*'}, mimetype='application/javascript')
+    else:
+        return app.response_class(data, status, {'Access-Control-Allow-Origin': '*'}, mimetype='application/json')
+
+def bad_request(message=None, exception=None):
+    if exception is not None:
+        message = exception.message
+    app.logger.info("Sending 400 Bad Request from client: {x}".format(x=message))
```

Comment:
Consider a magic like https://github.com/DOAJ/doaj/blob/develop/portality/view/account.py#L178 here. Doesn't cost anything to make a UUID. Best to give the magic to the client as well though! So when they say "I got error with magic X", we find it in a flash.